### ISSUE TITLE
The classic videogame bug happened!

### ISSUE THREAD

**Issue by Jeffrey-P-McAteer** [Jeffrey-P-McAteer/hackathon-2022-April#1]
Turns out the latest high-resolution nav-mesh approach works great for defining playspace as a set of boxes:

```
window.allowed_box_geometries = [
  [[2.4, 0.0, 6.4], [-5.75, 0.0, -2.2]], // covers large room
  [[1.2, 0.0, -2.7], [0.0, 0.0, -2.1]], // small zone where large room -> desk area door is
  [[2.4, 0.0, -7.4], [-0.8, 0.0, -2.6]], // covers desk area
  [[-4.05, 0.0, -3.0], [-4.6, 0.0, -2.2]], // small zone where bedroom / outside door is
  [[-1.8, 0.0, -6.0], [-4.6, 0.0, -2.8]], // remainder of empty "room" after player passes through door, z=-2.2 => -2.8 is an unpassable wall.
];
```

But the collision logic needs improvement, while testing this evening I noticed my player clipped through walls just by holding `W` and walking into them. Corners seemed to break the fastest.

tl;dr: [`crop_colliding_position_vector`](https://github.com/Jeffrey-P-McAteer/hackathon-2022-April/blob/master/www/app.js#L333) needs work to ensure players stay _inside_ the playable area.
